File: Code/1000/1512/1512.cpp
/******************************************************************************
 * 文件名: 1512
 * 作者: BOLUN XU
 * 创建日期: 2025
 * 版本: 1.0
 * 描述: 好数对的数目。
 * 时间复杂度： O(N ^2 )
 * 空间复杂度： O(1)
 ******************************************************************************/
#include <iostream>
#include <vector>

using namespace std;

class Solution {
  public:
    int numIdenticalPairs(vector<int>& nums) {
      int ans = 0;
      for (int i = 0; i < nums.size() - 1; ++i) {
        for (int j = i + 1; j < nums.size(); ++j) {
          if (nums[i] == nums[j]) {
            ++ans;
          }
        }
      }
      return ans;
    }
};

int main() {
  Solution s;
  vector<int> nums = {1, 2, 3, 1, 1, 3};
  cout << s.numIdenticalPairs(nums) << endl;
  return 0;
}

File: Code/1000/1512/1512.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
文件名: PersonalLeetCode
作者: Bolun Xu
创建日期: 2025
版本: 1.0
描述: 好数对的数目。
时间复杂度： O(N ^2 )
空间复杂度： O(1)
"""
from typing import List


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        ans = 0
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] == nums[j]:
                    ans += 1
        return ans


if __name__ == '__main__':
    sol = Solution()
    print(sol.numIdenticalPairs([1, 2, 3, 1, 1, 3]))
